### PR TITLE
next next next

### PR DESCRIPTION
**Pull request by SimonDanisch**
Addresses better convert and lazy shaders. Convert to OpenGL types as late as possible!

### CODE REVIEW

File: src/GLBuffer.jl [hunk 1]
```diff
@@ -45,10 +50,24 @@ end
...
-    unsafe_copy!(buffer, 1, newbuff, 1, length(buffer))
-    buffer.id   = newbuff.id
-    buffer.size = newbuff.size
+     newlength = newdims[1]
+     oldlen    = length(buffer)
+     if oldlen>0
+         old_data = gpu_data(buffer)
+     end
+    bind(buffer)
+    glBufferData(buffer.buffertype, newlength*sizeof(T), C_NULL, buffer.usage)
```
Comment (sjkelly) [inline src/GLBuffer.jl]: Weird indentation here.

**Comment by paulmthompson** [comment]
Somewhere in here the simple_triangle.jl example stopped working. I can adjust the color of the background, but I cannot get the triangle to appear. From what I can tell, the shaders are compiling and linking properly, the glDrawArrays function is starting with a 0 index as it should, and the prerender and postrender functions are being defined correctly. I'm not sure what is going on.

**Comment by SimonDanisch** [comment]
Ah yeah, haven't tested this for while...
I'll push a fix!

**Comment by paulmthompson** [comment]
Thanks so much! Sorry to bother you from multiple repos in the same day!

**Comment by SimonDanisch** [comment]
Okay! You need to checkout GLWindow, GeometryTypes and GLAbstraction... I pushed some other stuff to them while fixing this bug...
Was a really silly bug... The code in `GLRender.jl` is really old and I patched in a few new things a few weeks ago, which I didn't tidied up well enough. I should really give `GLAbstraction` some more love at some point!

**Comment by SimonDanisch** [comment]
No worries! If this helps to get a little closer to composable shaders that would already be all I could ask for :)

**Pull request merged**